nixos/modules/services/monitoring/riemann.nix

```diff
@@ -15,11 +15,18 @@ let
     [cfg.config] ++ (map (f: ''(load-file "${f}")'') cfg.configFiles)
   );
 
+  configFile = (if (cfg.configFile != null)
```

Comment:
I'll also suggest a cleaner alternative of implementing this:
```nix
{
  # ...
  config = mkIf cfg.enable {
    services.riemann.configFile = mkDefault (
      writeText "riemann-config.clj" riemannConfig
    );
  };
}
```

And then use `cfg.configFile` in the launcher.

The module system will then automatically take care of making a user-specified configFile a priority. This also has the advantage that the user can see the generated config file via `nix-instantiate --eval '<nixpkgs/nixos>' -A config.services.riemann.configFile`